lib/db_adapter.go

```diff
@@ -131,3 +132,76 @@ func (adapter *DbAdapter) GetPKIDForPublicKey(pkBytes []byte) *PKID {
 
 	return DBGetPKIDEntryForPublicKey(adapter.badgerDb, adapter.snapshot, pkBytes).PKID
 }
+
+//
+// AccessGroups
+//
+
+func (adapter *DbAdapter) GetAccessGroupEntryByAccessGroupId(accessGroupId *AccessGroupId) (*AccessGroupEntry, error) {
+	if accessGroupId == nil {
+		glog.Errorf("GetAccessGroupEntryByAccessGroupId: Called with nil accessGroupId, this should never happen")
+		return nil, nil
+	}
+
+	if adapter.postgresDb != nil {
+		pgAccessGroup := adapter.postgresDb.GetAccessGroupByAccessGroupId(accessGroupId)
+		if pgAccessGroup == nil {
+			return nil, nil
+		}
+		return pgAccessGroup.ToAccessGroupEntry(), nil
```

Comment:
you can update `ToAccessGroupEntry` to check if the pgAccessGroup is nil and we can simplify this block of code to be
`return pgAccessGroup.ToAccessGroupEntry()`